### DISCUSSION TITLE
TreeID not holding the metadata information for more than one run.

### DISCUSSION THREAD

**Discussion by nealresearch** [r-lidar/lidR#644] [Q&A]
I have a chm file and have ran a segment tree with watershed. I am trying to see the metadata for different treeID using the `filter_poi` function but if i run this once `las <- filter_poi(las, treeID == 1) ` and when i try to see the metadata for the second treeId  l

```
las <- filter_poi(las, treeID == 2) 
1: In min(x) : no non-missing arguments to min; returning Inf
2: In max(x) : no non-missing arguments to max; returning -Inf ")
```

Also the print out says everything is zero for the second run

```
class        : LAS (v1.4 format 6)
memory       : 5.5 Kb 
extent       : Inf, -Inf, Inf, -Inf (xmin, xmax, ymin, ymax)
coord. ref.  : ETRS89 / UTM zone 33N (N-E) 
area         : 0 m²
points       : 0  points
density      : 0 points/m²
```

I have somehow managed to append the data from the treeId in the list but i have to stick a while loop at the begining before all the process for creating the segementation. This is a problem because i dont the final number of treeID. Is there a faster solution for this

**Comment by Jean-Romain** [comment]
Please report a minimal reproducible example.

**Reply by nealresearch** [reply]
```r
library(raster)
library("lidR")
library(rgdal)
las <-  readLAS("E:\\las\\test.las")

chm <- raster(paste0("E:\\chm\\test.tif"))

ker <- matrix(1, 3, 3)
chm <- raster::focal(chm,  w = ker,  fun = mean,   na.rm = TRUE)
las <- segment_trees(las, watershed(chm, th_tree = 1, tol = 0.5, ext = 2))
las <- filter_poi(las, treeID == 1)
las
las1 <- filter_poi(las, treeID == 2)
las1
```

**Reply by nealresearch** [reply]
The las1 will give an empty metadata information if you run the whole code at once

**Reply by Jean-Romain** [reply]
You kept only the tree number 1 line 10. You cannot expect to find the tree number 2 line 12.
